checks/sast.go

```diff
@@ -25,7 +25,7 @@ import (
 // CheckSAST is the registered name for SAST.
 const CheckSAST = "SAST"
 
-var sastTools = map[string]bool{"github-code-scanning": true, "lgtm-com": true, "sonarcloud": true}
+var sastTools = map[string]bool{"github-code-scanning": true, "lgtm-com": true, "sonarcloud": true, "tfsec": true}
```

Comment:
this is fine on the condition that the repo only contains these sort of files, ie it's not both a development repo and a deployment repo.

@nadgowdas can you confirm the above is correct? Could you also provide us with some example repos with terraform files? Is there a way to differentiate between the two "kinds" of repos?

Reply:
Sorr I don't understand. Why can't it have both?

Reply:
it may have both, but it won't be reflected in the check results in the current implementation. The check uses an OR to pass, see https://github.com/ossf/scorecard/blob/d4dec7fdf1dd86238df448e0a4a116d5c36e056d/checks/sast.go#L162

Reply:
Yes, I agree with @laurentsimon 
I think we should have a pre-scanner that parses code from given repo and identifies applicable checks. E.g. if repo has '.tf` code, then we select `tfsec` to be applicable. If code contains both app code (e.g. .go and .tf) then we should select checks applicable for both types. 
Also, if repo contains `.yaml`/helm charts then we could have provision to run tools like [kubelinter](https://github.com/stackrox/kube-linter). 
I general, it can be made pluggable for any "kind" of artifact in the code repository. Thoughts ?

Reply:
makes sense to me. I suggested a similar approach in https://github.com/ossf/scorecard/pull/1981#discussion_r890245194: would be great if we can use the language API (https://docs.github.com/en/rest/repos/repos#list-repository-languages). If not we'll have to list files ourselves for `.tf`, etc.